src/components/PhotosHeader.vue

```diff
@@ -0,0 +1,202 @@
+			<Plus slot="icon" />
+			<ActionButton :close-after-click="true" :title="t('photos', 'Upload media')" @click="openUploader">
+				<FileUpload slot="icon" />
+			</ActionButton>
+			<ActionButton :close-after-click="true" :title="t('photos', 'Create new album')" @click="createNewAlbum">
+				<PlusBoxMultiple slot="icon" />
+			</ActionButton>
+		</Actions>
+		</Actions>
+
+		<template v-else>
+			<Actions :force-title="true" :menu-title="t('photos', 'Add to album')" :primary="true">
+				<ActionButton :close-after-click="true"
+					:primary="true"
+					:title="t('photos', 'Add to album')"
+					@click="openAlbumPicker">
+					<Plus slot="icon" />
+				</ActionButton>
+			</Actions>
+			<Actions :force-menu="true">
+				<ActionButton :close-after-click="true" :title="t('photos', 'Download')" @click="downloadSelection">
+					<DownloadOutline slot="icon" />
+				</ActionButton>
+				<ActionButton v-if="shouldFavorite"
+					:close-after-click="true"
+					:title="t('photos', 'Favorite')"
+					@click="favoriteSelection">
+					<Star slot="icon" />
+				</ActionButton>
+				<ActionButton v-else
+					:close-after-click="true"
+					:title="t('photos', 'Remove from favorites')"
+					@click="unFavoriteSelection">
+					<Star slot="icon" />
+				</ActionButton>
+				<ActionButton :close-after-click="true" :title="t('photos', 'Delete')" @click="deleteSelection">
+					<TrashCan slot="icon" />
+				</ActionButton>
+			</Actions>
+		</template>
+
+		<Loading v-if="loading > 0" class="loading-icon" />
+	</div>
+</template>
+<script>
+
+import { mapActions } from 'vuex'
+
+import Plus from 'vue-material-design-icons/Plus'
+import TrashCan from 'vue-material-design-icons/TrashCan.vue'
+import PlusBoxMultiple from 'vue-material-design-icons/PlusBoxMultiple'
+import FileUpload from 'vue-material-design-icons/FileUpload.vue'
+import Star from 'vue-material-design-icons/Star.vue'
+import DownloadOutline from 'vue-material-design-icons/DownloadOutline.vue'
+import Loading from 'vue-material-design-icons/Loading'
+
+import { Actions, ActionButton } from '@nextcloud/vue'
+
+import logger from '../services/logger.js'
+
+export default {
+	name: 'PhotosHeader',
+	components: {
+		Actions,
+		ActionButton,
+		Loading,
+		Plus,
+		TrashCan,
+		FileUpload,
+		PlusBoxMultiple,
+		Star,
+		DownloadOutline,
+	},
+	props: {
+		selection: {
+			type: Array,
+			default: () => [],
+		},
+	},
+
+	data() {
+		return {
+			loading: 0,
+		}
+	},
+
+	computed: {
+		/** @type {boolean} */
+		shouldFavorite() {
+			// Favorite all selection if at least one file is not on the favorites.
+			return this.selection.some((fileId) => this.$store.state.files.files[fileId].favorite === 0)
+		},
+	},
+
+	methods: {
+		...mapActions(['deleteFiles', 'toggleFavoriteForFiles', 'downloadFiles']),
+
+		openUploader() {
+
+		},
+
+		createNewAlbum() {
+
+		},
+
+		openAlbumPicker() {
+
+		},
+
+		async moveSelectionToAlbum() {
+
+		},
+
```

Comment:
```suggestion
```

Reply:
I'll keep those, but commented, so it can easily be reused in future work that should happen soon.